persistence/genesis.go

```diff
@@ -44,97 +45,72 @@ func (m *persistenceModule) populateGenesisState(state modules.PersistenceGenesi
-		if err != nil {
-			log.Fatalf("an error occurred inserting an app in the genesis state: %s", err.Error())
-		}
-		if err = addValueToPool(types.PoolNames_AppStakePool.String(), act.GetStakedAmount()); err != nil {
-			log.Fatalf("an error occurred inserting staked tokens into %s pool: %s", types.PoolNames_AppStakePool, err.Error())
-		}
-	}
-	for _, act := range state.GetNodes() {
-		addrBz, err := hex.DecodeString(act.GetAddress())
-		if err != nil {
-			log.Fatalf("an error occurred converting address to bytes %s", act.GetAddress())
-		}
-		pubKeyBz, err := hex.DecodeString(act.GetPublicKey())
-		if err != nil {
-			log.Fatalf("an error occurred converting pubKey to bytes %s", act.GetPublicKey())
-		}
-		outputBz, err := hex.DecodeString(act.GetOutput())
-		if err != nil {
-			log.Fatalf("an error occurred converting output to bytes %s", act.GetOutput())
-		}
-		err = rwContext.InsertServiceNode(addrBz, pubKeyBz, outputBz, false, StakedStatus, act.GetGenericParam(), act.GetStakedAmount(), act.GetChains(), act.GetPausedHeight(), act.GetUnstakingHeight())
-		if err != nil {
-			log.Fatalf("an error occurred inserting a service node in the genesis state: %s", err.Error())
-		}
-		if err = addValueToPool(types.PoolNames_ServiceNodeStakePool.String(), act.GetStakedAmount()); err != nil {
-			log.Fatalf("an error occurred inserting staked tokens into %s pool: %s", types.PoolNames_ServiceNodeStakePool.String(), err.Error())
-		}
-	}
-	for _, act := range state.GetFish() {
-		addrBz, err := hex.DecodeString(act.GetAddress())
-		if err != nil {
-			log.Fatalf("an error occurred converting address to bytes %s", act.GetAddress())
-		}
-		pubKeyBz, err := hex.DecodeString(act.GetPublicKey())
-		if err != nil {
-			log.Fatalf("an error occurred converting pubKey to bytes %s", act.GetPublicKey())
-		}
-		outputBz, err := hex.DecodeString(act.GetOutput())
-		if err != nil {
-			log.Fatalf("an error occurred converting output to bytes %s", act.GetOutput())
-		}
-		err = rwContext.InsertFisherman(addrBz, pubKeyBz, outputBz, false, StakedStatus, act.GetGenericParam(), act.GetStakedAmount(), act.GetChains(), act.GetPausedHeight(), act.GetUnstakingHeight())
-		if err != nil {
-			log.Fatalf("an error occurred inserting a fisherman in the genesis state: %s", err.Error())
-		}
-		if err = addValueToPool(types.PoolNames_FishermanStakePool.String(), act.GetStakedAmount()); err != nil {
-			log.Fatalf("an error occurred inserting staked tokens into %s pool: %s", types.PoolNames_FishermanStakePool.String(), err.Error())
-		}
+
+	stakedActorsInsertConfigs := []struct {
+		Name     string
+		Getter   func() []*coreTypes.Actor
+		InsertFn func(address []byte, publicKey []byte, output []byte, paused bool, status int32, serviceURL string, stakedTokens string, chains []string, pausedHeight int64, unstakingHeight int64) error
```

Comment:
```suggestion
		InsertFn func(address, publicKey, output []byte, paused bool, status int32, serviceURL, stakedTokens string, chains []string, pausedHeight, unstakingHeight int64) error
```

Reply:
I would say yes but if we want to enforce a style I would say let's be consistent and scout the whole codebase for similar instances and rectify accordingly otherwise it becomes a mix and match.

Another thing we could do is adding newlines when the signatures are particularly long. I think you did that somewhere in the codebase and I like it since it allows the reader to avoid horizontal scrolling.

Let me know how you want to go about your suggestion and, more generally, about styling guidelines, perhaps this could be related to linting? (#295)

Reply:
```
I would say yes but if we want to enforce a style I would say let's be consistent and scout the whole codebase for similar instances and rectify accordingly otherwise it becomes a mix and match.
```

Agreed. I was trying to be consistent with line 83 of this file (just local consistency).

I believe we can enforce not specifying types (when they're the same) using the automatic linters @okdas set-up since these are standard golang suggestions. They're currently off because we need to do a one-time cleanup.

You are correct that #295 encompasses this.

```
Another thing we could do is adding newlines when the signatures are particularly long. I think you did that somewhere in the codebase and I like it since it allows the reader to avoid horizontal scrolling.
```

I do like it but think it's overkill here because the function is anonymous.

```
Let me know how you want to go about your suggestion and, more generally, about styling guidelines, perhaps this could be related to linting? (https://github.com/pokt-network/pocket/issues/295)
```

Let's just be consistent between this and line 83 and #295 will encompass the work of enforcing repo-wide guidelines.